tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -151,11 +166,14 @@ def rules(cls):
       # B is both a RootRule and an intermediate product here.
       RootRule(B),
       RootRule(C),
+      RootRule(UnionX),
```

Comment:
I'm pretty sure this registration is not needed since the test does not construct a (root) `Param` of `UnionX`.